### PR TITLE
Build name Public Register page, added test

### PR DESCRIPTION
**Pull request by hemerem-ch**

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/model/ClientData.ts [hunk 1]
```diff
@@ -15,4 +15,7 @@ export interface ClientData {
     howIdentityDocsChecked?: string;
     whenIdentityChecksCompleted?: Date | string;
     confirmIdentityVerified?: string;
+    PublicRegisterName?: string;
+    PublicRegisterMiddleName?: string;
+    PublicRegisterLastName?: string;
```
Comment (cbroadley-ch) [inline src/model/ClientData.ts:20]: Can we change these variable names to start with lower case text?

File: src/controllers/personsPublicRegisterNameController.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const lang = selectLang(req.query.lang);
+    const errorList = validationResult(req);
+    if (!errorList.isEmpty()) {
+        const pageProperties = getPageProperties(formatValidationError(errorList.array(), lang));
+        const session: Session = req.session as any as Session;
+
+        const previousPageUrl: string = session?.getExtraData(PREVIOUS_PAGE_URL)!;
+        const previousPage = previousPageUrl === addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+            ? addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+            : addLangToUrl(BASE_URL, lang);
```
Comment (rnesbitt-ch) [inline src/controllers/personsPublicRegisterNameController.ts:51]: previous page here would be BASE_URL + USE_NAME_ON_PUBLIC_REGISTER

File: src/controllers/personsPublicRegisterNameController.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            payload: req.body,
+            ...pageProperties
+        });
+    } else {
+        const session: Session = req.session as any as Session;
+        const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
+
+        clientData.PublicRegisterName = req.body["first-name"];
+        clientData.PublicRegisterMiddleName = req.body["middle-names"];
+        clientData.PublicRegisterLastName = req.body["last-name"];
```
Comment (cbroadley-ch) [inline src/controllers/personsPublicRegisterNameController.ts:66]: Will need these changed to camel case to match 'ClientData' object structure

File: src/controllers/personsPublicRegisterNameController.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
+    const payload = {
+        "first-name": clientData.PublicRegisterName,
+        "middle-names": clientData.PublicRegisterMiddleName,
+        "last-name": clientData.PublicRegisterLastName
```
Comment (cbroadley-ch) [inline src/controllers/personsPublicRegisterNameController.ts:21]: Will need these changed to camel case to match 'ClientData' object structure

File: src/controllers/personsPublicRegisterNameController.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import { ClientData } from "../model/ClientData";
+import { getPreviousPageUrl } from "../services/url";
+import { USER_DATA, MATOMO_BUTTON_CLICK, PREVIOUS_PAGE_URL } from "../utils/constants";
+import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../utils/localise";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
```
Comment (ttingle-ch) [inline src/controllers/personsPublicRegisterNameController.ts:17]: we should not need to handle the null client data like this as it should never be null on this page.
it should be const clientData: ClientData = session.getExtraData(USER_DATA)!

File: src/controllers/personsPublicRegisterNameController.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        res.status(400).render(config.PERSONS_NAME_ON_PUBLIC_REGISTER, {
+            ...getLocaleInfo(locales, lang),
+            previousPage: previousPage,
+            currentUrl: BASE_URL + PERSONS_NAME_ON_PUBLIC_REGISTER,
+            payload: req.body,
+            ...pageProperties
+        });
+    } else {
+        const session: Session = req.session as any as Session;
+        const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
```
Comment (ttingle-ch) [inline src/controllers/personsPublicRegisterNameController.ts:62]: we should not need to handle the null client data like this as it should never be null on this page.
it should be const clientData: ClientData = session.getExtraData(USER_DATA)!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by rnesbitt-ch — APPROVED** [review]
LGTM

**Review by ttingle-ch — APPROVED** [review]
LGTM

**Pull request merged**